vsi/tests/it/main.rs

```diff
@@ -43,3 +51,81 @@ async fn run_succeeds() -> Result<()> {
 
     Ok(())
 }
+
+struct MemorySink {
+    // We don't care about performance here, mutex is not a concern.
+    buffer: Arc<Mutex<Vec<Artifact>>>,
+    id: Id,
+}
+
+#[async_trait]
+impl Sink for MemorySink {
+    async fn append_scan(&self, id: &Id, artifacts: Vec<Artifact>) -> Result<()> {
+        ensure!(id == &self.id, "id's did not match: {} <=> {}", id, self.id);
+        self.buffer.lock().await.extend(artifacts);
+        Ok(())
+    }
+}
+
+#[tokio::test]
+async fn archive_scan_produces_correct_prints() -> Result<()> {
+    let id = Id::new("Fake ID, I promise I'm 21.");
+    let sink = MemorySink {
+        buffer: Arc::new(Mutex::new(Vec::new())),
+        id: id.clone(),
+    };
+    let opts = Options::builder()
+        .root("tests/it/testdata/archive-scan")
+        .build();
+    let count = vsi::scan::artifacts(&sink, &id, opts)
+        .await
+        .context("vsi scan")?;
+    // two files in zip, plus the zip itself
+    assert_eq!(count, 3, "count of produced artifacts");
+
+    let results = Arc::try_unwrap(sink.buffer)
+        .expect("unwrap arc")
+        .into_inner();
+    assert_eq!(
+        count,
+        results.len(),
+        "reported count does not match sink ingestion count"
+    );
+
+    let result_map: HashMap<_, _> = results
+        .into_iter()
+        .map(|art| {
+            let (key, comb) = art.explode_string();
+            (key, comb.to_string())
+        })
+        .collect();
+
+    let pathsep = std::path::MAIN_SEPARATOR;
```

Comment:
🤔 I think we actually should be replacing `\` with `/` in paths... maybe that's why the windows integration test is broken. Feel free to leave that though, we can figure that out in the ticket that we have for fixing windows scans.